File: apps/mobile/modules/v12-native-controls/ios/V12NativeControlsModule.swift
import ExpoModulesCore

public final class V12NativeControlsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("V12NativeControls")

    View(V12HeaderButtonView.self) {
      Prop("label") { (view: V12HeaderButtonView, label: String) in
        view.setLabel(label)
      }
      Prop("systemImage") { (view: V12HeaderButtonView, systemImage: String) in
        view.setSystemImage(systemImage)
      }

      Events("onTriggered")
    }
  }
}

File: apps/mobile/modules/v12-native-controls/android/src/main/java/expo/modules/v12nativecontrols/V12NativeControlsModule.kt
package expo.modules.v12nativecontrols

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class V12NativeControlsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("V12NativeControls")

    View(V12HeaderButtonView::class) {
      Prop("label") { view: V12HeaderButtonView, label: String ->
        view.setLabel(label)
      }
      Prop("systemImage") { view: V12HeaderButtonView, systemImage: String ->
        view.setSystemImage(systemImage)
      }

      Events("onTriggered")
    }
  }
}
